### ISSUE TITLE
Class Specific Image Generation (L1 and L2 Regularization)

### ISSUE THREAD

**Issue by arjung128** [utkuozbulak/pytorch-cnn-visualizations#69]
Thank you so much for making this work open source! I really appreciate it.

I am trying to reproduce the results you've displayed. The code that is provided for this does very well for the no regularization case, but I cannot replicate the L1 and L2 regularization examples. How were these generated? Could the code for this also be kindly provided? 

Once again, thank you so much for your work!

**Comment by utkuozbulak** [comment]
Hey, those experiments were one of the first ones I tried (two years ago) so I doubt I still have the code. I'll try to recreate it and push it but no promises, and it is likely that I will not have time to do it soon.

**Comment by surekhag28** [comment]
@arjung128 can you please share the code because I am not getting expected results. Every time when I run the code (for different classes) I could see similar kind of images generating.

**Comment by arjung128** [comment]
@utkuozbulak I see. Please do comment on this thread (to notify me, and others) if you are able to find / replicate the code! Thanks!

**Comment by arjung128** [comment]
@surekhag28 I used the code [here](https://github.com/utkuozbulak/pytorch-cnn-visualizations/blob/master/src/generate_class_specific_samples.py) without any modifications. It seems to give me different results for the flamingo class and the spider class.

**Comment by surekhag28** [comment]
@arjung128 yes its giving for flamingo class but for other classes its almost similar.

**Comment by utkuozbulak** [comment]
By the way, for the spider and the snake example don't forget to change the architecture to VGG19, I also remember using some blurring the image every nth iteration to help aid the optimization. You can get more detail on how to aid the optimization from its paper. I suggest reading it.

**Comment by surekhag28** [comment]
@utkuozbulak which paper you are talking about?

**Comment by utkuozbulak** [comment]
https://arxiv.org/abs/1312.6034 and its extension (another paper) that I can't seem to find right now.

**Comment by surekhag28** [comment]
@utkuozbulak yes I read the same paper and there only I found that they are using softmax function to maximise the probability score which I could not find in your code and I am quite confused with it (raised issue for the same #71)

**Comment by utkuozbulak** [comment]
Then I suggest you first try to understand what you read before expecting any free code and wasting anyone elses time.

Quoting from the paper:
"It should be noted that we used the (unnormalised) class scores, rather than the class posteriors, returned by the soft-max layer." They specifically say that they do not use softmax.

**Comment by utkuozbulak** [comment]
@arjung128 @surekhag28 I found the paper I was referring to two comments above. You can follow the procedures listed in this paper: 
http://yosinski.com/media/papers/Yosinski__2015__ICML_DL__Understanding_Neural_Networks_Through_Deep_Visualization__.pdf in order to improve the quality of the generated visualizations. Have a look at Section 3. Good luck.
